pkg/authorization/reverse_permission_claims_authorizer_test.go

```diff
@@ -0,0 +1,330 @@
+		attr         authorizer.Attributes
+		claim        *apisv1alpha1.AcceptablePermissionClaim
+		wantDecision authorizer.Decision
+		wantReason   string
+		wantError    string
+	}{
+		{
+			name: "allowed: unclaimed group/resource",
+			attr: authorizer.AttributesRecord{Verb: "create", Resource: "secrets", Namespace: "foo", Name: "bar"},
+			claim: &apisv1alpha1.AcceptablePermissionClaim{
+				PermissionClaim: apisv1alpha1.PermissionClaim{
+					GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+					All:           true,
+					Verbs:         apisv1alpha1.Verbs{RestrictTo: []string{"get", "list"}},
+				},
+			},
+			wantDecision: authorizer.DecisionNoOpinion,
+			wantReason:   "unclaimed resource in API binding name=\"binding\"",
+		},
+		{
+			name: "allowed: rejected claim",
+			attr: authorizer.AttributesRecord{Verb: "create", Resource: "configmaps", Namespace: "foo", Name: "bar"},
+			claim: &apisv1alpha1.AcceptablePermissionClaim{
+				PermissionClaim: apisv1alpha1.PermissionClaim{
+					GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+					All:           true,
+					Verbs:         apisv1alpha1.Verbs{RestrictTo: []string{"get", "list"}},
+				},
+				State: apisv1alpha1.ClaimRejected,
+			},
+			wantDecision: authorizer.DecisionAllow,
+			wantReason:   "claim is not accepted in API binding name=\"binding\"",
+		},
+		{
+			name: "allowed: unclaimed resource, not matching namespace",
+			attr: authorizer.AttributesRecord{Verb: "create", Resource: "configmaps", Namespace: "foo", Name: "bar"},
+			claim: &apisv1alpha1.AcceptablePermissionClaim{
+				PermissionClaim: apisv1alpha1.PermissionClaim{
+					GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+					ResourceSelector: []apisv1alpha1.ResourceSelector{
+						{Name: "", Namespace: "baz"},
+						{Name: "", Namespace: "buzz"},
+					},
+					Verbs: apisv1alpha1.Verbs{RestrictTo: []string{"get", "list"}},
+				},
+				State: apisv1alpha1.ClaimAccepted,
+			},
+			wantDecision: authorizer.DecisionAllow,
+			wantReason:   "resource doesn't match any resource selector in API binding name=\"binding\"",
+		},
+		{
+			name: "allowed: unclaimed resource, matching namespace, not matching name",
+			attr: authorizer.AttributesRecord{Verb: "create", Resource: "configmaps", Namespace: "baz", Name: "bar"},
+			claim: &apisv1alpha1.AcceptablePermissionClaim{
+				PermissionClaim: apisv1alpha1.PermissionClaim{
+					GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+					ResourceSelector: []apisv1alpha1.ResourceSelector{
+						{Name: "foo", Namespace: "baz"},
+						{Name: "foo", Namespace: "buzz"},
+					},
+					Verbs: apisv1alpha1.Verbs{RestrictTo: []string{"get", "list"}},
+				},
+				State: apisv1alpha1.ClaimAccepted,
+			},
+			wantDecision: authorizer.DecisionAllow,
+			wantReason:   "resource doesn't match any resource selector in API binding name=\"binding\"",
+		},
+		{
+			name: "allowed: unclaimed verb",
```

Comment:
eh, this is very bad naming, my bad, it was a shortcut for "we didn't claim this verb because it's in the set of restrictTo verbs". I'll change the test name to `allowed: claimed resource but verb is restricted to` (doesn't sound great either).